### PR TITLE
Update chpl_arch.py to handle 64-bit ARM, KNL, and Skylake

### PR DESCRIPTION
**Pull request by dmk42**
ARM presents a challenge for `chpl_arch.py` because Intel architectures are accessed with the `-march=` compiler flag, and ARM architectures are accessed with the `-mcpu=` flag.  This change includes an indication of which flag to use in the output of `chpl_arch.py` and updates it to handle the relevant ARM architectures.

This allowed the removal of some special cases from `clangUtil.cpp`.

Also, KNL and Skylake have been added.

Makefiles had to be changed to expect `CHPL_MAKE_TARGET_BACKEND_ARCH` to include part of the flag.  Some Python scripts had to be changed to stop after the first `=` when parsing environment variables.

This PR addresses issues #5686 and #8820 .

### CODE REVIEW

**Comment by dmk42** [comment]
@ronawho - Would you please review this?

**Comment by dmk42** [comment]
I'm labeling this a work in progress until we figure out how we want to handle splitting the (arch|cpu)=architecture strings.

**Comment by dmk42** [comment]
@ronawho - This is now reimplemented to return a tuple and verified working on a wide variety of architectures, so it's ready for re-review.

File: util/chplenv/chpl_comm_substrate.py [hunk 1]
```diff
@@ -15,7 +15,6 @@ def get():
     if not substrate_val:
         comm_val = chpl_comm.get()
         platform_val = chpl_platform.get('target')
-        arch_val = chpl_arch.get('target', get_lcd=True)
```
Comment (ronawho) [inline util/chplenv/chpl_comm_substrate.py]: Why is this here?

Comment (ronawho) [reply util/chplenv/chpl_comm_substrate.py]: Oh, nevermind. I see it's dead code  (that I missed removing in https://github.com/chapel-lang/chapel/commit/29ec9bce3d2b30cd15d84c2d0d07e8c3b806f5d0#diff-45c544a18723ce3d378babf320393a0fL28)

**Comment by ronawho** [comment]
Overall, this looks good to me. 

I wonder if we could use a shorter name than `CHPL_TARGET_BACKEND_ARCH_FLAG`. Maybe just `CHPL_TARGET_ARCH_FLAG`

I wonder if it would be cleaner to use a named tuple. And that way you could do things like: `chpl_arch.get('target').arch` instead of manually detupling

@ben-albrecht may be interested in reviewing some of the chplenv infrastructure changes (though probably not the chpl_arch.py changes)

File: util/chplenv/third_party_utils.py [hunk 2]
```diff
@@ -15,10 +15,11 @@
 #
 @memoize
 def default_uniq_cfg_path():
+    (flag, arch) = chpl_arch.get('target', map_to_compiler=True,
```
Comment (ronawho) [inline util/chplenv/third_party_utils.py]: I would go with `(_, arch)` anywhere the `flag` isn't going to be used. Either that or maybe have chpl_arch return a named tuple so that you could just do `chpl_arch.get(...).arch`

**Comment by ben-albrecht** [comment]
> @ben-albrecht may be interested in reviewing some of the chplenv infrastructure changes (though probably not the chpl_arch.py changes)

LGTM
